cub/cub/util_type.cuh

```diff
@@ -85,16 +80,19 @@ CUB_NAMESPACE_BEGIN
 #ifndef _CCCL_DOXYGEN_INVOKED // Do not document
 namespace detail
 {
-//! Alias to the given iterator's value_type.
-// Aliases to std::iterator_traits, since users can specialize this template to provide traits for their iterators. We
-// only defer to the libcu++ implementation for NVRTC.
-template <typename Iterator>
-using value_t =
-#  if _CCCL_COMPILER(NVRTC)
-  typename ::cuda::std::iterator_traits<Iterator>::value_type;
-#  else // ^^^ _CCCL_COMPILER(NVRTC) ^^^ // vvv !_CCCL_COMPILER(NVRTC) vvv
-  typename std::iterator_traits<Iterator>::value_type;
-#  endif // !_CCCL_COMPILER(NVRTC)
+using ::cuda::std::iter_difference_t;
+using ::cuda::std::iter_reference_t;
+using ::cuda::std::iter_value_t;
```

Comment:
This is a bug. Previously, `value_t<T>` would support iterators with a `value_type` of `void`. `::cuda::std::iter_value_t` does not support this any longer.

Reply:
Reverted back to using `iterator_traits`.